### PR TITLE
Fixes #27045 - Refactor BreadcrumbSwitcher to PF-React

### PR DESCRIPTION
**Pull request by glekner**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -93,7 +93,7 @@
     "multiselect": "~0.9.12",
     "number_helpers": "^0.1.1",
     "patternfly": "^3.58.0",
-    "patternfly-react": "^2.34.1",
+    "patternfly-react": "^2.36.1",
```
Comment (glekner) [inline package.json]: the switcher wasn't exported in pf's `index.js` in the original PR.  it was fixed since then so we need a bump

**Review by xprazak2 — DISMISSED** [review]
Works for me.

**Comment by xprazak2** [comment]
packaging: https://github.com/theforeman/foreman-packaging/pull/3854

**Comment by glekner** [comment]
@amirfefer @ohadlevy can you test?

**Comment by ohadlevy** [comment]
I gave it a quick spin, and it does looks working well overall, one annoying item i found was that my browser was trying to auto complete the filter input, perhaps we shouldn't call it "name" and then the browser won't try to autocomplete?

**Review by amirfefer — DISMISSED** [review]
Works well, just wondering if it make sense to move `BreadcrumbBar` component to patternfly as well

**Comment by amirfefer** [comment]
@ohadlevy  - I've opened a PR that should fix the autocomplete issue

**Review by amirfefer — COMMENTED** [review]
@glekner the autocomplete fix has been merged in [PF](https://github.com/patternfly/patternfly-react/pull/2407), can you update pf version in this PR ?

**Comment by glekner** [comment]
@amirfefer React requires camelCase for DOM attributes, we need to change `autocomplete` to `autoComplete` in PF.

[PR](https://github.com/patternfly/patternfly-react/pull/2457)

**Comment by glekner** [comment]
autoComplete PR got merged, bumped PF version.
need to update packaging PR aswell if you have time @xprazak2

**Comment by xprazak2** [comment]
I updated the packaging PR.

**Comment by glekner** [comment]
- rebased

**Comment by glekner** [comment]
- rebased and updated vendor packages to support latest PF-React
This PR is ready

**Comment by ezr-ondrej** [comment]
We have lost the translations of the list items... I do not feel it is as important, but I would like to have it addressed... how do you think we should approach it @glekner?
I can think of forcing the translations on the server side, do you think it would solve the issue?

**Comment by glekner** [comment]
> We have lost the translations of the list items... I do not feel it is as important, but I would like to have it addressed... how do you think we should approach it @glekner?
> I can think of forcing the translations on the server side, do you think it would solve the issue?

I added translations.

**Comment by glekner** [comment]
[test travis]

File: webpack/assets/javascripts/react_app/components/BreadcrumbBar/BreadcrumbBar.js [hunk 2]
```diff
@@ -68,9 +69,9 @@ class BreadcrumbBar extends React.Component {
           {isSwitchable && (
             <BreadcrumbSwitcher
               open={isSwitcherOpen}
-              isLoadingResources={isLoadingResources}
+              isLoading={isLoadingResources}
               hasError={hasError}
-              resources={resourceSwitcherItems}
+              resources={translateObject(resourceSwitcherItems)}
```
Comment (ezr-ondrej) [inline webpack/assets/javascripts/react_app/components/BreadcrumbBar/BreadcrumbBar.js]: `resourceSwitcherItems` is an array of objects, and thats expected by `BreadcrumbSwitcher` it seems.
Now it seems we are translating the whole object If I am reading it correctly.
I guess we eighter need to translate in `BreadcrumbBarActions -> loadSwitcherResourcesByResource -> formatResults` or create new helper `translatePropOfArrayOfObjects(arrayOfObjects, <[props_to_translate]>)`

**Review by ezr-ondrej — CHANGES_REQUESTED** [review]
Tests are failing, what's I guess related to the below comment. Please feel free to ping me on IRC once done, I would love to get this one in :)

File: webpack/assets/javascripts/react_app/components/BreadcrumbBar/BreadcrumbBar.js (hunk 2 above)
Comment (glekner) [reply webpack/assets/javascripts/react_app/components/BreadcrumbBar/BreadcrumbBar.js]: whats wrong with translating the whole object. btw is translating even the desired approach here? isn't it risky/unnecessary? @tbrisker

**Comment by glekner** [comment]
- rebased

File: webpack/assets/javascripts/react_app/components/BreadcrumbBar/BreadcrumbBar.js (hunk 2 above)
Comment (ezr-ondrej) [reply webpack/assets/javascripts/react_app/components/BreadcrumbBar/BreadcrumbBar.js]: The problems are:

- You are providing array `[{id: .., name: ..},{....}]` to the translations, but the method expects object, and it translates all the values in the object. so in here, it is translating the arrays.
- We don't want to translate anything else from the inner object, just the names.

**Comment by ezr-ondrej** [comment]
The travis failures are related, the translations has been done before on the client, I believe we should do the same.

File: webpack/assets/javascripts/react_app/components/BreadcrumbBar/BreadcrumbBar.js (hunk 2 above)
Comment (tbrisker) [reply webpack/assets/javascripts/react_app/components/BreadcrumbBar/BreadcrumbBar.js]: you need to translate each string separately, but not resource names - for example, "Compute Resources" should be translated but "my-libvirt-CR" should not.

**Comment by glekner** [comment]
@ezr-ondrej moved translation process to the actions file

**Review by ezr-ondrej — APPROVED** [review]
Thanks a lot @glekner! Great contribution, nice cleanup.
Merging!

**Pull request merged**